go/oasis-node/cmd/common/metrics/metrics.go

```diff
@@ -19,50 +19,47 @@ import (
 )
 
 const (
-	cfgMetricsMode              = "metrics.mode"
-	cfgMetricsAddr              = "metrics.address"
-	cfgMetricsPushJobName       = "metrics.push.job_name"
-	cfgMetricsPushInstanceLabel = "metrics.push.instance_label"
-	cfgMetricsPushInterval      = "metrics.push.interval"
-
-	metricsModeNone = "none"
-	metricsModePull = "pull"
-	metricsModePush = "push"
+	CfgMetricsMode     = "metrics.mode"
+	CfgMetricsAddr     = "metrics.address"
+	CfgMetricsLabels   = "metrics.labels"
+	CfgMetricsJobName  = "metrics.job_name"
+	CfgMetricsInterval = "metrics.interval"
+
+	MetricsModeNone = "none"
+	MetricsModePull = "pull"
+	MetricsModePush = "push"
 )
 
 // Flags has the flags used by the metrics service.
 var Flags = flag.NewFlagSet("", flag.ContinueOnError)
 
-// ServiceConfig contains the configuration parameters for metrics.
-type ServiceConfig struct {
-	// Mode is the service mode ("none", "pull", "push").
-	Mode string
-	// Address is the address of the push server.
-	Address string
-	// JobName is the name of the job for which metrics are collected.
-	JobName string
-	// InstanceLabel is the instance label of the job being collected for.
-	InstanceLabel string
-	// Interval defined the push interval for metrics collection.
-	Interval time.Duration
-}
-
-// GetServiceConfig gets the metrics configuration parameter struct.
-func GetServiceConfig() *ServiceConfig {
-	return &ServiceConfig{
-		Mode:          viper.GetString(cfgMetricsMode),
-		Address:       viper.GetString(cfgMetricsAddr),
-		JobName:       viper.GetString(cfgMetricsPushJobName),
-		InstanceLabel: viper.GetString(cfgMetricsPushInstanceLabel),
-		Interval:      viper.GetDuration(cfgMetricsPushInterval),
+// GetStringMapString is a drop-in replacement for viper.GetStringMapString due to https://github.com/spf13/viper/issues/608.
+func GetStringMapString(val string) map[string]string {
```

Comment:
As mentioned in my earlier comment, consider moving this under the common flags package.